File: app/src/main/java/com/ed/redditapp/lib/TextChangedWatcher.java
package com.ed.redditapp.lib;

import android.text.Editable;
import android.text.TextWatcher;

public interface TextChangedWatcher extends TextWatcher {
    default void beforeTextChanged(CharSequence s, int start, int count, int after) {}
    default void afterTextChanged(Editable s) {}
    default void onTextChanged(CharSequence s, int start, int before, int count) {
        textChanged(s.toString());
    }

    void textChanged(String text);
}

File: app/src/main/java/com/ed/redditapp/lib/TextChangedWatcher.kt
package com.ed.redditapp.lib

import android.text.Editable
import android.text.TextWatcher

interface TextChangedWatcher : TextWatcher {
    override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}
    override fun afterTextChanged(s: Editable) {}
    override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
        textChanged(s.toString())
    }

    fun textChanged(text: String)
}